### ISSUE TITLE
MEET | As a user of the app I want to see the noi contact.

### ISSUE THREAD

**Issue by sseppi** [noi-techpark/it.bz.noi.community.android#97]

**Comment by chiaraDimension** [comment]
@sseppi 
As regards point 1, can you confirm that we don't have to add this filter to the query (as reported [here](https://github.com/noi-techpark/it.bz.noi.community-api/blob/development/docs/consumer.md))? 

`$filter=Microsoft.Dynamics.CRM.ContainValues(PropertyName%3D%40p1,PropertyValues%3D%40p2)%20and%20statuscode%20eq%201`

Using this filter, the query returns only 4 accounts.

Query:
`https://api.community.noi.testingmachine.eu/accounts?@p1='crb14_accountcat_placepresscommunity'&@p2=['952210000']&$select=name,noi_nameit,telephone1,address1_composite&$filter=Microsoft.Dynamics.CRM.ContainValues(PropertyName%3D%40p1,PropertyValues%3D%40p2)%20and%20statuscode%20eq%201&$count=true`

Response:
[accounts_with_filter.txt](https://github.com/noi-techpark/it.bz.noi.community.android/files/8713883/accounts_with_filter.txt)

Without the filter, we have 2804 accounts instead.

Query:
`https://api.community.noi.testingmachine.eu/accounts?@p1='crb14_accountcat_placepresscommunity'&@p2=['952210000']&$select=name,noi_nameit,telephone1,address1_composite&$count=true`

Response:
[Uploading accounts_without_filter.txt…]()

Is the second query the right one?

**Comment by sseppi** [comment]
@chiaraDimension 

No, the right one is the first query. At the moment you are making the query to the testing environment where we have only a few companies marked as NOI community.

Sorry it was my mistake, the filter to not consider inactive accounts and contact is needed as reported in the link you shred.

**Comment by chiaraDimension** [comment]
> @chiaraDimension
> 
> No, the right one is the first query. At the moment you are making the query to the testing environment where we have only a few companies marked as NOI community.
> 
> Sorry it was my mistake, the filter to not consider inactive accounts and contact is needed as reported in the link you shred.

Ok! Is it possibile to mark other companies as NOI community in the testing environment as well? It's really difficult to test the implementation and especially the performance if we don't have a real number of accounts.

**Comment by sseppi** [comment]
We deployed in also the production environment that you can use.
You cand find more detail at the following issue:
https://github.com/noi-techpark/it.bz.noi.community-api/issues/19

**Comment by chiaraDimension** [comment]
@sseppi 
How should be handle contacts without associated company (_parentcustomerid_value = null)?

i.e.
`{
            "@odata.etag": "W/\"70373695\"",
            "emailaddress1": "l.geier@noi.bz.it",
            "firstname": "Lisa",
            "lastname": "Geier",
            "_parentcustomerid_value": null,
            "contactid": "7f3ebb76-7a6f-ea11-a811-000d3a4aa602"
        },`

**Comment by MarE1991** [comment]
@sseppi @chiaraDimension there are no contacts without a company or institution associated. If so, then it's an error. We check all Community contacts every few days and add the missing information. So if this situation happens, then I would also say that we just don't show the info.

**Comment by chiaraDimension** [comment]
> @sseppi @chiaraDimension there are no contacts without a company or institution associated. If so, then it's an error. We check all Community contacts every few days and add the missing information. So if this situation happens, then I would also say that we just don't show the info.

The example above was from testing machine. Effectively, in production environment I don't find contacts without associated company. In this case, there is another problem instead: just a few contacts find a match with a company in companies list (in particular, they seem to be only contacts of NOI Labs). Can you check?

**Comment by MarE1991** [comment]
@chiaraDimension indeed, there are exactly 21 contacts that have a LAB as account and though as company. In these cases, the company name would be the parent account (see image attached). @sseppi Is it possible to create an API where the parent account is taken in consideration in case there is one? Otherwise, I have to confront me with our CRM repersentative and solve the problem with these 21 contacts. THX
![image](https://user-images.githubusercontent.com/92152188/169816953-239129ea-8e09-4b00-81c8-39578898bba2.png)

**Comment by chiaraDimension** [comment]
@MarE1991 Maybe we had a misunderstanding. We already have the information to retrieve the company name of the contact. What we think is wrong is that we have the company name for only these 21 contacts. For everyone else, the parent account id doesn't match any company.

For example:
- for `Giovanni Carabin`, we show `NOI Labs - Field Robotics Lab`
- for `Stefano Seppi`, we don't show any company name

@sseppi these are the query, can you confirm?

ACCOUNTS:
`https://api.community.noi/.{{host}}/accounts?@p1='crb14_accountcat_placepresscommunity'&@p2=['952210000']&$select=name,noi_nameit,telephone1,address1_composite&$filter=Microsoft.Dynamics.CRM.ContainValues(PropertyName%3D%40p1,PropertyValues%3D%40p2)%20and%20statuscode%20eq%201&$count=true`
CONTATTI:
`https://api.community.noi/.{{host}}/contacts?$filter=Microsoft.Dynamics.CRM.ContainValues(PropertyName%3D%40p1,PropertyValues%3D%40p2)%20and%20statuscode%20eq%201&@p1='noi_contactcat_placepresscommunity'&@p2=['181640000']&$select=emailaddress1,firstname,lastname,fullname,_parentcustomerid_value&$count=true`

where `host`=
- `opendatahub.bz.it` for production environment
- `testingmachine.eu` for testing machine environment

Then, to match a contact with its account we use the field `_parentcustomerid_value` (we search the account with `accountid` equals to this `_parentcustomerid_value`).

It almost seems that, using the filter `$filter=Microsoft.Dynamics.CRM.ContainValues(PropertyName%3D%40p1,PropertyValues%3D%40p2)%20and%20statuscode%20eq%201` we don't obtain all the necessary accounts.

**Comment by MarE1991** [comment]
@chiaraDimension ok, now I got it. This is actually wrong, we have to take the "Company Name"-information and only use the "Parent Account"-one when there is one (means when the Company Name is a LAB). @sseppi can you please verify this? THX

**Comment by Piiit** [comment]
@MarE1991 No usually you just need to update, see the link under https://github.com/noi-techpark/it.bz.noi.community.android#deployment ... click that on your phone and see if you get an "update" button inside the Google Play Store

**Comment by Piiit** [comment]
@MarE1991 If that did not solve your issues, please hold the app icon for some time and choose "App Info" or a similar item... then delete the cache or application data, restart the app and see...

**Comment by MarE1991** [comment]
@Piiit now it worked, thank you!! 👍 For me the new page is perfectly fine, but I would discuss this in our meeting this afternoon, then we can close the issue.
